Carbon.IIS/Functions/Set-CIisCollectionItem.ps1

```
@@ -0,0 +1,261 @@
+                Set-Variable -Name 'keyValueWritten' -Value $true -Scope 1
+            }
+
+            Write-Information $Message
+        }
+
+        $firstLine = 'IIS configuration collection '
+        $firstLineWritten = $false
+        $keyValueWritten = $false
+
+        $save = $false
+        $collectionArgs = @{}
+
+        $elementPath = ''
+        if ($ConfigurationElement)
+        {
+            $firstLine = "${firstLine}$($ConfigurationElement.ElementTagName)"
+            $collectionArgs['ConfigurationElement'] = $ConfigurationElement
+            $elementPath = $ConfigurationElement.ElementTagName
+            if (Get-Member -Name 'SectionPath' -InputObject $ConfigurationElement)
+            {
+                $elementPath = $ConfigurationElement.SectionPath
+            }
+        }
+        else
+        {
+            $displayPath = Get-CIisDisplayPath -SectionPath $SectionPath `
+                                               -LocationPath $locationPath `
+                                               -SubSectionPath $CollectionName
+            $firstLine = "${firstLine}${displayPath}"
+            $elementPath = $SectionPath
+            $collectionArgs['SectionPath'] = $SectionPath
+            if ($LocationPath)
+            {
+                $collectionArgs['LocationPath'] = $LocationPath
+            }
+        }
+
+        if ($CollectionName)
+        {
+            $elementPath = "${elementPath}/$($CollectionName)"
+            $collectionArgs['Name'] = $CollectionName
+        }
+
+        $collection = Get-CIisCollection @collectionArgs
+
+        if (-not $collection)
+        {
+            return
+        }
+
+        $keyAttrName = Get-CIisCollectionKeyName -Collection $collection
+
+        if (-not $keyAttrName)
+        {
+            $msg = "Unable to find key for ${elementPath}."
+            Write-Error -Message $msg
+            return
+        }
+
+    }
+
+    process
+    {
+        if (-not $collection)

```

We'll never get here because this condition is already checked in the `begin` block.

---

PowerShell still executes `process` and `end` blocks when there's an error. If someone ignores/silently continues the error, the `process` and `end` blocks will also execute.

---

Oh, TIL.